library/src/blas1/rocblas_iamaxmin_template.h

```diff
@@ -0,0 +1,225 @@
+#pragma once
+
+#include "rocblas_batched_arrays.h"
+
+#ifndef AMAX_AMIN_REDUCTION
+#error undefined macro AMAX_AMIN_REDUCTION
+#endif
+
+#define QUOTE2(S) #S
+#define QUOTE(S) QUOTE2(S)
+
+#define JOIN2(A, B) A##B
+#define JOIN(A, B) JOIN2(A, B)
+
+//
+//
+// Template on the reduction types.
+//
+//
+template <typename T>
+struct reduction_types
+{
+public:
+    using Ti = T;
+    using To = T;
+};
+
+//
+// Specialization of the reduction types for the float complex.
+//
+template <>
+struct reduction_types<rocblas_float_complex>
+{
+public:
+    using Ti = rocblas_float_complex;
```

Comment:
`public` is redundant in `struct`.

Reply:
public is default in struct, not redundant.

Reply:
but done.